Automattic/pocket-casts-android
# [Paywall Experiment] Add new featured cards instead of the old ones

## Description
- Adds the new featured cards in the paywall experiment
- Updates the Plus pill to have gradient effect

Fixes #2631

## Testing Instructions

1. Apply this [paywall.patch](https://github.com/user-attachments/files/16612993/paywall.patch) and run in `debug`
2. Have `PAYWALL_AB_EXPERIMENT` feature flag enabled
3. Open paywall
4. ✅ Ensure plus pill has gradient effect
5. ✅ The new featured cards are displayed added here https://github.com/Automattic/pocket-casts-android/pull/2660

Try with different screen sizes and font sizes

## Screenshots or Screencast 

| Regular Device | Small Device |
|--------|--------|
| <img src="https://github.com/user-attachments/assets/cea97c62-ae69-4e00-9d39-07d0d9d4dcdb" width="400px"> | <img src="https://github.com/user-attachments/assets/a35b9d14-1a34-4745-991a-372879d6ae06"  width="400px"> |



## Checklist
 
#### I have tested any UI changes...
